Add details on versioning and deprecation (#94)
ansible-collections/overview: collection_requirements.rst

Versioning and deprecation
~~~~~~~~~~~~~~~~~~~~~~~~~~

> * Every ansible minor version series (2.10.x) will keep the major version of a collection constant. So if ``community.general`` 1.2.0 is included in 2.10.0, 2.10.x will contain the latest ``community.general`` 1.y.z release that was available when building 2.10.x. If ``community.general`` 2.0.0 is released, it will **not** be included in ansible 2.10.x, but only in 2.11.0 (if it was published before that was released).

Reviewer: ```suggestion
* To preserve backward compatibility for users, every ansible minor version series (2.10.x) will keep the major version of a collection constant. If ansible 2.10.0 includes ``community.general`` 1.2.0, then each 2.10.x release will include the latest ``community.general`` 1.y.z release available at build time. Ansible 2.10.x will never include a ``community.general`` 2.y.x release, even if it is available. Major collection version changes will be included in the next ansible minor release (2.11.0, 2.12.0, and so on).
```

* Therefore, please make sure that the current major release of your collection included in 2.10.0 receives at least bugfixes as long new 2.10.x releases are produced.
* Since new minor releases are included, you can include new features, modules and plugins. You must make sure that you do not break backwards compatibility! (See `semantic versioning <https://semver.org/>`_.) This means in particular:

>   * You can fix bugs in patch releases, but not add new features or deprecate things.

Reviewer 2: I think you need a blank line when changing bullet-list indent levels
```suggestion

  * You can fix bugs in patch releases, but not add new features or deprecate things.
```

  * You can add new features and deprecate things in minor releases, but not remove things or change behavior of existing features.

>   * Removals and breaking changes can only be made in major releases.

Reviewer: ```suggestion
  * You can only remove things or make breaking changes in major releases.
```
